src/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
-if (SANITIZE_ADDRESS)
+if (ENABLE_SANITIZERS)
+    set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=undefined -fsanitize=leak")
+    set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=undefined -fsanitize=leak")
     add_definitions(-DBOOST_USE_ASAN)
```

Comment:
Use target_compile_options/definitions instead. Maybe in a macro target_enable_sanitizers, on in wrapper macro for add_library/executable. Global variables like CMAKE_CXX_FLAGS and commands like add_definitions to be avoided.